src/ring_theory/roots_of_unity.lean

```diff
@@ -599,6 +599,23 @@ begin
   rwa ring_char.eq_iff.mpr h
 end
 
+lemma ne_one {ζ : R} (hζ : is_primitive_root ζ k) (hk : 1 < k) : ζ ≠ 1 :=
+hζ.pow_ne_one_of_pos_of_lt zero_lt_one hk ∘ (pow_one ζ).trans
```

Comment:
Maybe this should be moved earlier? Perhaps nearby https://leanprover-community.github.io/mathlib_docs/ring_theory/roots_of_unity.html#is_primitive_root.pow_ne_one_of_pos_of_lt and https://leanprover-community.github.io/mathlib_docs/ring_theory/roots_of_unity.html#is_primitive_root.one

Reply:
Done.